### PR TITLE
fix: (#12016) adding redirect back feature from insights table

### PR DESCRIPTION
**Pull request by Arun-chaitanya**
## Problem

Redirect back to the dashboard after editing an insight edited from dashboard #12016

## Changes

Added redirection back to insights table if user clicked on edit insight from insight table list.

👉 *Stay up-to-date with [PostHog coding conventions](https://posthog.com/docs/contribute/coding-conventions) for a smoother review.*

### CODE REVIEW

**Review by Twixes — CHANGES_REQUESTED** [review]
Unfortunately this doesn't actually resolve the issue because it adds returning to the Insights page (which I don't believe is needed), and not returning to the previously-viewed Dashboard.

If you're eager to rework this for dashboards, then I have a suggestion regarding the experience: it would be worth indicating that you're going to be moved away from the insight on save. Maybe with setting the `sideIcon` on the save button to `<IconArrowLeft />` in that case (this would also apply to "Save and add to insight").

**Comment by Arun-chaitanya** [comment]
@Twixes I totally get it now. I thought the issue was regarding coming back to the insights list page if we click `edit` from there. I now realise that we have many dashboards where we can see the insights, and we need to redirect back to those specific dashboards after the "Save". Please comment and let me know if I am thinking it correctly now.

**Comment by Twixes** [comment]
Yes @Arun-chaitanya, specifically to the dashboard from which the user just arrived at the insight. I think the URL-based approach makes sense, since it survives a page refresh. However, it's important that this behavior is clearly signaled.

**Comment by Arun-chaitanya** [comment]
Yes, @Twixes I too think the above approach is right, but just that we will have to add this second argument in `urls.insightEdit()` from all sorts of dashboards. It is easy as we can do global search for `urls.insightEdit(`. But I was wondering if kea has any plugin which can return us the present url.
And also any suggestion from your side on how to signal this behaviour clearly. I think tooltip on Save button should do the trick.

**Comment by Twixes** [comment]
Kea has the `kea-router` plugin, which exports a `router` logic – that's what we usually use for handling location URLs and location history. As for signaling, a no-delay tooltip sounds good.

**Comment by Arun-chaitanya** [comment]
@Twixes done the required changes. Please review the changes once

File: frontend/src/scenes/insights/InsightPageHeader.tsx [hunk 1]
```diff
@@ -267,14 +268,24 @@ export function InsightPageHeader({ insightLogicProps }: { insightLogicProps: In
...
-                                insightChanged={insightChanged || queryChanged}
-                            />
+                            <Tooltip
+                                title={
+                                    currentLocation.searchParams?.from
+                                        ? 'You will be redirected to your previous page'
+                                        : undefined
+                                }
+                            >
+                                <div>
```
Comment (Twixes) [inline frontend/src/scenes/insights/InsightPageHeader.tsx:278]: This should definitely be _within_ `InsightSaveButton`, not wrapping it

File: frontend/src/scenes/insights/insightLogic.ts [hunk 2]
```diff
@@ -674,7 +674,10 @@ export const insightLogic = kea<insightLogicType>([
             dashboardsModel.actions.updateDashboardInsight(savedInsight)
 
             const mountedInsightSceneLogic = insightSceneLogic.findMounted()
-            if (redirectToViewMode) {
+            const redirectionPath = decodeParams(router.values.location.search, '?')
```
Comment (Twixes) [inline frontend/src/scenes/insights/insightLogic.ts:677]: This looks like an open redirect vulnerability. I suggest narrowing this mechanism down to just dashboards. We have a `highlightInsightId` search param in `dashboardLogic`, we could use a similar `originDashboardId` param here. Then we'd call `urls.dashboard(...)` here, avoiding free-form redirects (which can be used maliciously)

**Review by Twixes — CHANGES_REQUESTED** [review]
Hmm, it looks like this still pertains to the insights list, and not dashboards. I don't think this is a concern with the dashboards list, because that is usually navigated to specifically choose an insight to hone in on. With dashboards, there _is_ a workflow of tweaking insights while thinking of the dashboard overall, which is why the redirect would be useful _there_.

File: frontend/src/scenes/insights/insightLogic.ts (hunk 2 above)
Comment (Arun-chaitanya) [reply frontend/src/scenes/insights/insightLogic.ts:677]: Got it. Will check.

**Pull request closed without merging**